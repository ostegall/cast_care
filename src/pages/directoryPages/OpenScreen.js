import { Text, View, Pressable } from 'react-native';
import styles from '../../styles';

export default function OpenScreen({navigation}) { //Liability Page for user to accept to reduce client's liability for injuries/nonaction
  return (
    <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
      <Text style={styles.page_text}>
          *Insert Liability Statement Here*
      </Text>
      <Text>
          Please Accept to Continue
      </Text>
      <Pressable style={styles.button} 
            onPress = {() => {
                navigation.navigate('Home'); //navigates to screen for diagnosing cast issue(s)
            }}>
        <Text style={styles.text}>
            Accept
        </Text>
      </Pressable>    
    </View>
  )
}