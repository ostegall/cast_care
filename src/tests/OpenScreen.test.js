import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import OpenScreen from '../pages/directoryPages/OpenScreen';

//Makes sure app component renders correctly without crashing
it("renders without crashing", () => {
  shallow(<OpenScreen />);
});

//Snapshot test to ensure UI stays consistent
it('renders correctly', () => {
  const tree = renderer.create(<OpenScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});