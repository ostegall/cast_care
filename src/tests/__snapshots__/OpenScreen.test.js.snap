// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    *Insert Liability Statement Here*
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    Please Accept to Continue
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4292c7",
        "borderRadius": 10,
        "elevation": 2,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "minWidth": 88,
        "paddingLeft": 16,
        "paddingRight": 16,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.35,
        "shadowRadius": 5,
        "width": 120,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 14,
          "letterSpacing": 0.25,
          "textAlign": "center",
        }
      }
    >
      Accept
    </Text>
  </View>
</View>
`;
